lib/ansible/modules/cloud/amazon/ec2_vpc_egress_igw.py

```diff
@@ -0,0 +1,200 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This Ansible library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ec2_vpc_egress_igw
+short_description: Manage an AWS VPC Egress Only Internet gateway
+description:
+    - Manage an AWS VPC Egress Only Internet gateway
+version_added: "2.5"
+author: Daniel Shepherd (@shepdelacreme)
+options:
+  vpc_id:
+    description:
+      - The VPC ID for the VPC that this Egress Only Internet Gateway should be attached.
+    required: true
+  state:
+    description:
+      - Create or delete the EIGW
+    default: present
+    choices: [ 'present', 'absent' ]
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Ensure that the VPC has an Internet Gateway.
+# The Internet Gateway ID is can be accessed via {{eigw.gateway_id}} for use in setting up NATs etc.
+ec2_vpc_egress_igw:
+  vpc_id: vpc-abcdefgh
+  state: present
+register: eigw
+
+'''
+
+RETURN = '''
+gateway_id:
+    description: The ID of the Egress Only Internet Gateway or Null.
+    returned: always
+    type: string
+    sample: eigw-0e00cf111ba5bc11e
+vpc_id:
+    description: The ID of the VPC to attach or detach gateway from.
+    returned: always
+    type: string
+    sample: vpc-012345678
+'''
+
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    camel_dict_to_snake_dict
+)
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be picked up by HAS_BOTO3 in AnsibleAWSModule
+
+
+def delete_eigw(module, conn, eigw_id):
+    """
+    Delete EIGW.
+
+    module     : AnsibleModule object
+    conn       : boto3 client connection object
+    eigw_id    : ID of the EIGW to delete
+    """
+    changed = False
+
+    try:
+        response = conn.delete_egress_only_internet_gateway(DryRun=module.check_mode, EgressOnlyInternetGatewayId=eigw_id)
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        # When boto3 method is run with DryRun=True it returns an error on success
+        # We need to catch the error and return something valid
+        if e.response.get('Error', {}).get('Code') == "DryRunOperation":
```

Comment:
BotoCoreError does not have a .response attribute so this may throw an AttributeError.

Reply:
Ok I updated both sections that would be affected by this.